File: Plus One/Plus One.java
public class Solution {
    public int[] plusOne(int[] digits) {
        Stack<Integer> res = new Stack<Integer>();
        int carry = 1;
        for(int i = digits.length - 1; i >= 0; --i){
            int num = (carry + digits[i]) % 10;
            carry = (carry + digits[i]) / 10;
            res.push(num);
        }
        if(carry != 0){
            res.push(carry);
        }
        
        int len = res.size();
        int[] ret = new int[len];
        for(int i = 0; i < len; ++i){
            ret[i] = res.pop();
        }
        
        return ret;
    }
}

File: Plus One/Plus One.cpp
class Solution {
public:
    vector<int> plusOne(vector<int> &digits) {

        bool overflow = true;
        for(int i = digits.size() - 1; i >= 0; --i){
            if(digits[i] < 9){
                digits[i]++;
                overflow = false;
                break;
            }
            else{
                digits[i] = 0;
            }
        }
        if(overflow){
            digits.push_back(0);
            for(unsigned int i = 0; i < digits.size(); ++i){
                if(!i)      digits[i] = 1;
                else        digits[i] = 0;
            }
        }
        return digits;

    }
};
